File: scratch/test_ftp.js
const ftp = require("basic-ftp")

async function testFtp() {
    const client = new ftp.Client()
    client.ftp.verbose = true
    try {
        console.log("Connecting via basic-ftp with Explicit TLS (secure: true)...")
        await client.access({
            host: "WIN8236.site4now.net",
            user: "tspmasterprd",
            password: "tspmasterprd",
            secure: true,
            secureOptions: {
                rejectUnauthorized: false
            }
        })
        console.log("=================> SUCCESSFUL FTPS LOGIN VIA BASIC-FTP!")
        const list = await client.list()
        console.log("Directory listing:", list)
    } catch (err) {
        console.log("FTPS Error:", err)
    }
    client.close()
}

testFtp()

File: scratch/test_ftp.py
import ftplib

host = 'WIN8236.site4now.net'
user = 'tspmasterprd'
passwd = 'tspmasterprd'

try:
    print("Connecting to FTP_TLS...")
    ftps = ftplib.FTP_TLS()
    ftps.connect(host, 21, timeout=10)
    print("Sending auth()...")
    ftps.auth()
    print("Sending PBSZ 0 & PROT P before login...")
    ftps.sendcmd('PBSZ 0')
    ftps.sendcmd('PROT P')
    print("Logging in with user/pass...")
    res = ftps.login(user, passwd)
    print(f"=================> SUCCESS! Response: {res}")
    ftps.retrlines('LIST')
    ftps.quit()
except Exception as e:
    print(f"FAILED: {e}")
